### ISSUE TITLE
no-unmodified-loop-condition false positive

### ISSUE THREAD

**Issue by mourner** [eslint/eslint#5445]
Using ESLint 2.2.0, the following code causes a false positive:

```js
/*eslint no-unmodified-loop-condition: 2 */
var a = [1, 2, 3];
var len = a.length;
for (var i = 0; i < len - 1; i++) {}
// produces: `len` is not modified in this loop. (no-unmodified-loop-condition)
```

This is a false positive because `i` is modified in the loop condition. If you replace `len - 1` with `len`, the error disappears.

Reproducible in the [ESLint demo](http://eslint.org/demo/) too. The issue is similar to #5166, but not the same. cc @feross @mysticatea

**Comment by alberto** [comment]
Confirmed

**Comment by mourner** [comment]
@mysticatea thank you for fixing this so quickly!
